tests/test_backend.py

```diff
@@ -0,0 +1,441 @@
+        "HttpRequestError when the exception originates in a downloader "
+        "middleware."
+    ),
+    strict=True,
+)
+@inlineCallbacks
+def test_additional_requests_unhandled_downloader_middleware_exception():
+    items = []
+
+    with MockServer(EchoResource) as server:
+
+        @attr.define
+        class ItemPage(ItemWebPage):
+            http_client: HttpClient
+
+            async def to_item(self):
+                try:
+                    await self.http_client.request(
+                        server.root_url,
+                        body=b'raise',
+                    )
+                except HttpError as e:
+                    return {'exc': e.__class__}
+
+        class TestDownloaderMiddleware:
+            def process_response(self, request, response, spider):
+                if b'raise' in response.body:
+                    raise RuntimeError
+                return response
+
+        class TestSpider(Spider):
+            name = 'test_spider'
+            start_urls = [server.root_url]
+
+            custom_settings = {
+                'DOWNLOADER_MIDDLEWARES': {
+                    TestDownloaderMiddleware: 1,
+                    'scrapy_poet.InjectionMiddleware': 543,
+                },
+            }
+
+            async def parse(self, response, page: ItemPage):
+                item = await page.to_item()
+                items.append(item)
+
+        crawler = make_crawler(TestSpider, {})
+        yield crawler.crawl()
+
+    assert items == [{'exc': HttpError}]
+
+
+@inlineCallbacks
+def test_additional_requests_dont_filter():
+    """Verify that while duplicate regular requests are filtered out,
+    additional requests are not (neither relative to the main requests not
+    relative to each other).
+
+    In Scrapy, request de-duplication is implemented on the scheduler, and
+    because additional requests do not go through the scheduler, this works as
+    expected.
+    """
+    items = []
+
+    with MockServer(EchoResource) as server:
+
+        @attr.define
+        class ItemPage(ItemWebPage):
+            http_client: HttpClient
+
+            async def to_item(self):
+                await self.http_client.request(server.root_url)
+                await self.http_client.request(server.root_url)
+                return {'foo': 'bar'}
+
+        class TestSpider(Spider):
+            name = 'test_spider'
+
+            custom_settings = {
+                'DOWNLOADER_MIDDLEWARES': {
+                    'scrapy_poet.InjectionMiddleware': 543,
+                },
+            }
+
+            def start_requests(self):
+                yield Request(server.root_url)
+                yield Request(server.root_url)
+
+            async def parse(self, response, page: ItemPage):
+                item = await page.to_item()
+                items.append(item)
+
+        crawler = make_crawler(TestSpider, {})
+        yield crawler.crawl()
+
+    assert items == [{'foo': 'bar'}]
```

Comment:
it seems this test doesn't check that 2 requests in to_item are made, it could pass if the second (or both) of the requests were no-op, without raising an exception